runners/flink/src/test/java/org/apache/beam/runners/flink/PortableStateExecutionTest.java

```diff
@@ -86,6 +86,7 @@ public void testExecution() throws Exception {
     options.setRunner(CrashingRunner.class);
     options.as(FlinkPipelineOptions.class).setFlinkMaster("[local]");
     options.as(FlinkPipelineOptions.class).setStreaming(isStreaming);
+    options.as(FlinkPipelineOptions.class).setParallelism(4);
```

Comment:
Good find. We should always set the parallelism to cap resource consumption. I would prefer 2 if the intention is to exercise parallel execution path, 1 otherwise.

Reply:
Our current test setup wastes quite a few resources because it uses the number of available cores in most places. Will do a follow-up to fix that.

>I would prefer 2 if the intention is to exercise parallel execution path, 1 otherwise.

Do you think a slightly higher parallelism would increase the chance of exposing bugs? For example, the key distribution might be more skewed with a higher parallelism.

Reply:
To give a meaningful signal, the test should probably be different/specialized and assert something about the parallel execution? If not, then we can probably just set it to 1 ? As it stands we are operating on hope :)

Reply:
Testing to a large degree is based on hope to expose bugs while testing functionality :) We have seen tests running fine with parallelism 1 in the past while they fail with parallelism >1. ValidatesRunner for instance still runs with parallelism 1. But I'm ok with lowering the parallelism to 2.